Quantum-Control-Applications-QuAM/Superconducting/quam_libs/quam_builder/machine.py

```diff
@@ -142,6 +144,26 @@ def add_octaves(machine: QuAM, octaves_settings: Dict, quam_state_path: Union[Pa
     return machine
 
 
+def add_external_mixers(machine: QuAM, quam_state_path: Union[Path, str]):
+    if isinstance(quam_state_path, str):
+        quam_state_path = Path(quam_state_path)
+    quam_state_path = str(quam_state_path.parent.resolve())
+
+    for wiring_by_element in machine.wiring.values():
+        for wiring_by_line_type in wiring_by_element.values():
+            for references in wiring_by_line_type.values():
+                for reference in references:
+                    if "mixers" in references.get_unreferenced_value(reference):
+                        frequency_converter = FrequencyConverter(
```

Comment:
Is `machine.mixers[mixer_name]` supposed to be pointing towards a frequency converter?
It might make more sense to attach the frequency converter, mixer, etc to the channel, rather than to a top-level entry such as `mixers`. The reason is that this way the local oscillator frequency could reference the channel / qubit `LO_frequency` using the parent